File: app/src/main/java/com/shenghaiyang/oksign/library/LibraryAdapter.java
package com.shenghaiyang.oksign.library;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import com.shenghaiyang.oksign.R;
import java.util.List;

final class LibraryAdapter extends RecyclerView.Adapter<LibraryViewHolder> {

  private final LayoutInflater inflater;
  private final List<Library> libraries;

  LibraryAdapter(Context context, List<Library> libraries) {
    this.inflater = LayoutInflater.from(context);
    this.libraries = libraries;
  }

  @Override public LibraryViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    return new LibraryViewHolder(inflater.inflate(R.layout.library_item, parent, false));
  }

  @Override public void onBindViewHolder(LibraryViewHolder holder, int position) {
    Library library = libraries.get(position);
    holder.nameView.setText(library.name);
    holder.licenseNameView.setText(library.licenseName);
    holder.licenseContentView.setText(library.licenseContent);
  }

  @Override public int getItemCount() {
    return libraries.size();
  }
}

File: app/src/main/java/com/shenghaiyang/oksign/library/LibraryAdapter.kt
package com.shenghaiyang.oksign.library

import android.support.v7.widget.RecyclerView
import android.view.ViewGroup

class LibraryAdapter(val libraries: List<Library>) : RecyclerView.Adapter<LibraryViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = LibraryViewHolder.create(parent)

    override fun getItemCount() = libraries.size

    override fun onBindViewHolder(holder: LibraryViewHolder, position: Int) {
        holder.bind(libraries[position])
    }

}
